### PR TITLE
Add ONNX Export Support for torch.scalar_tensor

### PR DESCRIPTION
**Pull request by lara-hdr**
Support exporting torch.scalar_tensor() to ONNX.
This will allow making operations on dynamic scalars (like x.size(dim) where x is a tensor of dynamic shape) and exporting them to ONNX.

This is a dummy example of operations that could not be exported dynamically before this PR:

```
size_x = x.size(0)
size_y = y.size(0)
size_x_y_static = torch.tensor([size_x , size_y])  # size_x_y_static is traced as constant

size_x = torch.scalar_tensor(size_x).unsqueeze(0)
size_y = torch.scalar_tensor(size_y).unsqueeze(0)
size_x_y_dynamic = torch.cat((size_x , size_y))  # size_x_y_dynamic is dynamic and depends on x and y's size
```

### CODE REVIEW

File: test/onnx/test_pytorch_onnx_onnxruntime.py [hunk 1]
```diff
@@ -185,6 +185,22 @@ def forward(self, x):
...
+                    torch.scalar_tensor(input.size(1), dtype=torch.int64)
+
+        x = torch.randn(2, 3, 4)
+        y = torch.randn(7, 8, 9)
+        model = test()
+        self.run_test(model, x, test_with_inputs=[y],
+                      input_names=['input_1'],
+                      output_names=['output_1', 'output_2'],
+                      dynamic_axes={'input_1': [0, 1, 2],
+                                    'output_1': [0],
```
Comment (BowenBao) [inline test/onnx/test_pytorch_onnx_onnxruntime.py]: For outputs they don't need to be dynamic in shape right? Are both the outputs here rank 0 tensor (scalar)?

**Review by BowenBao — APPROVED** [review]
LGTM

**Review by houseroad — APPROVED** [review]
LGTM
